### PR TITLE
Vspm passthrough

### PR DESCRIPTION
**Pull request by aanisov**

### CODE REVIEW

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd-salvator-x-m3.cfg [hunk 1]
```diff
@@ -58,7 +58,7 @@ dt_passthrough_nodes = [
     "/sound",
     "/vga-encoder",
     "/vga",
-    "/vspm_if",
+#    "/vspm_if",
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd-salvator-x-m3.cfg]: remove, do not comment

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-doma.dts [hunk 2]
```diff
@@ -44,5 +52,62 @@
...
+		compatible = "simple-bus";
+		ranges;
+		#address-cells = <2>;
+		#size-cells = <2>;
+
+		vspb: vspm@fe960000 {
+			compatible = "renesas,vspm";
+			reg = <0 0xfe960000 0 0x8000>, <0 0xfe96f000 0 0x200>;
+			interrupts = <GIC_SPI 266 IRQ_TYPE_LEVEL_HIGH>;
+/*			clocks = <&cpg CPG_MOD 626>, <&cpg CPG_MOD 607>;
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-doma.dts]: either remove or put a comment why

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts [hunk 3]
```diff
@@ -774,6 +786,7 @@
 	status = "okay";
 
 	iommus = <&ipmmu_ds1 34>;
+	power-domains = <&sysc R8A7795_PD_A3VP>;
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts]: At least you have to add a comment describing why do you put power-domain here the sdhi is not even belongs to.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-domd.dts [hunk 4]
```diff
@@ -338,10 +338,22 @@
...
 	clock-names = "du.0", "du.1", "du.2", "du.3",
 				  "lvds.0", "dclkin.0", "dclkin.1",
-				  "dclkin.2", "dclkin.3", "3dge";
+				  "dclkin.2", "dclkin.3", "3dge",
+				  "vspbc", "fcpvbc",
+				  "vspbd", "fcpvbd",
+				  "vspi0", "fcpvi0",
+				  "vspi1", "fcpvi1",
+				  "fdp0", "fcpf0",
+				  "fdp1", "fcpf1";
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-domd.dts:357]: I haven't seen corresponding code in rcar-du driver to parse these clocks. You get names from the const array directly. Why do you need to keep clocks here?

Comment (aanisov) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-domd.dts:357]: The correspondent code in rcar-du is located right below your gfx related code.

Comment (aanisov) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-domd.dts:357]: I actually duplicated your code. With a minor exception that I track list of clocks both in H3 and M3 SoCs but enable only clocks available for a SoC we are running on.

Comment (otyshchenko1) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-domd.dts:357]: Now I see. Please treat my question as non-actual.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd-salvator-x-m3.cfg (hunk 1 above)
Comment (aanisov) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd-salvator-x-m3.cfg]: done

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-doma.dts (hunk 2 above)
Comment (aanisov) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-doma.dts]: wiped out

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts (hunk 3 above)
Comment (aanisov) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts]: done

**Comment by aanisov** [comment]
new [patch](https://github.com/xen-troops/meta-xt-prod-ces2018/pull/60/commits/d099f15ae9d1793a782749659fe63779b685817) is added, please consider

**Review by andr2000 — APPROVED** [review]

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts [hunk 5]
```diff
@@ -774,6 +786,13 @@
 	status = "okay";
 
 	iommus = <&ipmmu_ds1 34>;
+
+	/*
+	   FIXME: following is a hacky way to get A3VP power domain enabled from
+	  DomD. SDHI2 is choosen because onboard eMMC is going to be turned on
+	  always and overriding its native ALWAYS_ON power domain is not harmful
+	 */
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-dom0.dts:794]: This comment must be present for 7796 as well. Everywhere whether this hack is present

**Pull request merged**